test/merge-patient-lab-iips-mocha.js

```diff
@@ -1,174 +1,263 @@
  
-    it("should be an object on nodeInstance", function() {
+    it("should have have the expected node facade", function() {
         var node = test.createComponent(compFactory);
-        expect(node).to.be.an('object');
+
+        node.should.be.an('object');
+        node.should.include.keys( 'nodeName', 'componentName', 'inPorts', 'outPorts',
+                                   'deleteAllVnis', 'deleteVni', 'vni', 'vnis' );
+
+        node.inPorts.should.be.an('object'); 
+        node.outPorts.should.be.an('object'); 
+
+        node.deleteAllVnis.should.be.a('function');
+        node.deleteVni.should.be.a('function');
+        node.vni.should.be.a('function');
+        node.vnis.should.be.an('object');
     });
 
-    it("should have a vni function", function() {
+    it( "should have have expected input & output ports", function() {
         var node = test.createComponent(compFactory);
-        expect(node.vni).to.be.a('function');
+
+        node.inPorts.should.be.an('object'); 
+        node.inPorts.should.have.all.keys( 'patient', 'labwork' );
+        node.inPorts.patient.should.be.an('object');
+        node.inPorts.labwork.should.be.an('object');
+
+        node.outPorts.should.be.an('object'); 
+        node.outPorts.should.have.all.keys( 'output', 'error' );
+        node.outPorts.output.should.be.an('object');
+        node.outPorts.error.should.be.an('object');
     });
 
-    describe('#vni', function() {
+    describe( '#vni', function() {
 
-        it("should return a valid vni object", function() {
+        it( "should return a vni facade", function() {
             var node = test.createComponent(compFactory);
             var vni = node.vni();
+
             expect(vni).to.be.an('object');
+            vni.should.have.all.keys( 'vnid', 'delete', 'errorState', 
+                                      'inputStates', 'outputState', 'node' );
+
+            vni.vnid.should.be.a('string');
+            vni.delete.should.be.a('function');
+            vni.errorState.should.be.a('function');
+            vni.inputStates.should.be.a('function');
+            vni.outputState.should.be.a('function');
+            vni.node.should.be.an('object');
         });
 
-        it("should have inputStates function", function() {
-            var node = test.createComponent(compFactory);
-            expect(node.vni().inputStates).to.be.a('function');
-        });
+        describe( '#errorState', function() {
 
-        it("should have outputState function", function() {
-            var node = test.createComponent(compFactory);
-            node.vni().outputState.should.exist;
-            node.vni().outputState.should.be.a('function');
-        });
+            it( "should initially return empty error state", function() {
+                var node = test.createComponent(compFactory);
+                var errorState = node.vni().errorState();
+                errorState.should.have.all.keys( 'vnid', 'data', 'error', 'lm' );
```

Comment:
I disagree.  Object is  generic.  By having the keys, we are sure we actually have something that looks like an error object.  Let's discuss.